### PR TITLE
Add xpra display-manager

### PR DESCRIPTION
**Pull request by offlinehacker**
###### Motivation for this change

This pull request adds xpra as display-manager, which allows forwarding of x11 windows over ssh/tcp/sockets... While this pull request does not support full desktops, i don't need that in first place and support for that can be added later.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by offlinehacker** [comment]
I have not managed, to get pulseaudio forwarding working. It shows among pulseaudio recording and playback devices on server and client, but when i start chromium, there's no sound. As this is not so critial for me, i will solve this problem another time.

File: nixos/modules/services/x11/display-managers/xpra.nix [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    services.xserver.deviceSection = ''
+      VideoRam 192000
+    '';
+
+    services.xserver.displayManager.job = {
+      logsXsession = true;
+
+      execCmd = ''
+        ${optionalString (cfg.pulseaudio)
+          "export PULSE_COOKIE=/var/run/pulse/.config/pulse/cookie"}
```
Comment (offlinehacker) [inline nixos/modules/services/x11/display-managers/xpra.nix]: Yes

**Comment by offlinehacker** [comment]
I will be doing more testing in following days and add some more usefull options(encryption). For now it works fine with my vbox dev vms.

**Comment by bjornfor** [comment]
@offlinehacker: Have you tested that this does not break simple ```xpra shadow ssh:myserver```? I have to ask, because you are removing XPRA_LOG_DIR which I explicitly added in 6d48539fdb25b0bf2772352614707a9f23628e7e to make xpra run.

Btw, what does the xpra module give beyond ```xpra shadow ...```?

**Comment by offlinehacker** [comment]
I think this variable should be added in user environment variables, as
this is runtime environment speciffic? or at least somehow make it optional.

**Comment by offlinehacker** [comment]
@bjornfor as opposed to xpra shadow, I don't want to run any other display manager/desktop besides xpra

**Comment by NodeGuy** [comment]
@offlinehacker I'm curious to know what you're using this for.

**Comment by offlinehacker** [comment]
@NodeGuy i'm using it instead of display manager, so i can connect on it with xpra client and have separated windows for each application in a secure manner, not whole new desktop manager. I'm running it in my vm-s.

**Comment by NodeGuy** [comment]
Cool, thanks.

**Comment by offlinehacker** [comment]
I'm merging this, as i'm using xpra as display manager for a long time without any issues.

**Comment by bjornfor** [comment]
@offlinehacker: But you are breaking "xpra shadow ssh:myserver"...?

**Comment by offlinehacker** [comment]
@bjornfor but `XPRA_LOG_DIR ` should not be set in package, this is runtime variable that you can set in user profile or system wide with `environment.variables`. I will still check if I can make default point to home and override on xserver start.

**Pull request merged**